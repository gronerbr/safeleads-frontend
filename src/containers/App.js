import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Example from '../components/Example/Example'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Hello React World</h1>
        </header>
        <Example />
      </div>
    );
  }
}

export default App;
