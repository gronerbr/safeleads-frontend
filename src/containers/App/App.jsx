import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles';
import 'normalize.css';
import 'font-awesome/css/font-awesome.min.css';
import style from './App.scss';
import Routes from '../Routes/Routes';

const theme = createMuiTheme({
  palette: {
    primary: {
      light: '#d05ce3',
      main: '#9c27b0',
      dark: '#6a0080',
      contrastText: '#fff',
    },
    secondary: {
      light: '#6effff',
      main: '#00e5ff',
      dark: '#00b2cc',
      contrastText: '#000',
    },
  },
});

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasError: false,
    };
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true });
    // eslint-disable-next-line
    console.log('componentDidCatch info ', info);
    // eslint-disable-next-line
    console.log('componentDidCatch error ', error);
    // You can also log the error to an error reporting service (GA or other)
    // logErrorToMyService(error, info);
  }

  render() {
    if (this.state.hasError) {
      // eslint-disable-next-line
      console.error('THERE\'S AN ERROR IN APPLICATION, PLEASE FIX IT!');
    }

    return (
      <div className={style.App} ref={1}>
        <main>
          <MuiThemeProvider theme={theme}>
            <Routes />
          </MuiThemeProvider>
        </main>
      </div>
    );
  }
}

App.contextTypes = {
  router: PropTypes.object.isRequired,
};

export default App;
