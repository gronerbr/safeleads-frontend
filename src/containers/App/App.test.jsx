import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import App from './App';
import createStore from '../../redux/create';

const store = createStore();

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store} key="provider">
      <App />
    </Provider>,
    div,
  );
});
