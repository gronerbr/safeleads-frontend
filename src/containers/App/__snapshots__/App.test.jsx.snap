// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className={undefined}
>
  <header
    className={undefined}
  >
    <img
      alt="logo"
      className={undefined}
      src="logo.svg"
    />
    <h1
      className={undefined}
    >
      Hello React World
    </h1>
  </header>
  <main>
    This project accepts arrow function in classes
    <div>
      <p>
        It
        '
        s an example of React component. You can navigate to  
        <a
          href="/counter"
          onClick={[Function]}
        >
          Counter
        </a>
          to see another page.
      </p>
    </div>
  </main>
</div>
`;
